helios-services/src/main/java/com/spotify/helios/agent/TaskRunner.java

```diff
@@ -176,8 +176,12 @@ private int run0() throws InterruptedException, DockerException {
             .build().newScheduler();
 
         while (!healthChecker.get().check(containerId)) {
-          // TODO(negz): Why aren't we using getContainerInfo() here?
-          final ContainerState state = docker.inspectContainer(containerId).state();
+          final ContainerState state = getContainerState(containerId);
+          if (state == null) {
+            log.warn("container state was null during health checking: {}: {}",
+                     config, containerId);
+            throw new RuntimeException("container state was null during health checking");
```

Comment:
a better error message for both the log line and exception would be something like `"container <containerId> was not found during health checking, or has no State object"` - as-is the ID of the container is not included and it's not totally clear what it means for the state to be null